atsign-foundation/Atsign_OpenWrt_packages
# fix: Working Makefile for csshnpd on MIPS

Following work for https://github.com/atsign-foundation/noports/issues/1189

**- What I did**

* Removed python packages, as they were causing trouble
* Refined Makefile for C sshnpd

**- How I did it**

Iterations using the [GL.iNet SDK](https://github.com/gl-inet/sdk) to target an MT-1300 with stock firmware.

**- How to verify it**

Install Python2 (for Debian12 temporarily add `deb http://archive.debian.org/debian/ stretch contrib main non-free` to `/etc/apt/sources.list` then `sudo apt update && sudo apt install -y python 2.7`)
Install the [GL.iNet SDK](https://github.com/gl-inet/sdk) per the README
Pick a target platform (e.g. MT-1300) and download the target for that (e.g. `./download.sh ramips-1907`)
`cd sdk/1907/ramips`
`cp feeds.conf.default feeds.conf`
Add `src-git atsign https://github.com/atsign-foundation/Atsign_OpenWRT_packages^4143d489379d4fa1c1a9b6f631c1dd163b36230b` to `feeds.conf`
`./scripts/feeds update -f`
`./scripts/feeds install csshnpd`
`mv ./staging_dir/host/bin/cmake ./staging_dir/host/bin/cmake-3.15.1`
Ensure cmake 3.24 or later is installed (e.g. `sudo apt install cmake`)
Then copy it into place (e.g.):
```sh
cp $(which cmake) ./staging_dir/host/bin/cmake
cp -R /usr/share/cmake-3.25 ./staging_dir/host/share/
```
Build the package: `make V=s package/csshnpd/compile`

`./bin/packages/mipsel_24kc/atsign/csshnpd_0.1.6-1_mipsel_24kc.ipk` should now be ready.

NB there will be a whole lot of `cc1: note: someone does not honour COPTS correctly, passed 0 times` but these don't seem to cause any harm

**- Description for the changelog**

fix: Working Makefile for csshnpd on MIPS
